src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Organizers/RegisterOrganizerValidator.cs

```diff
@@ -0,0 +1,80 @@
+
+namespace TicketManager.Core.Services.Endpoints.Organizers;
+
+public class RegisterOrganizerValidator: Validator<RegisterOrganizerRequest>
+{
+    private readonly IServiceScopeFactory scopeFactory;
+    private readonly MockableCoreDbResolver dbResolver;
+    public RegisterOrganizerValidator(IServiceScopeFactory scopeFactory, MockableCoreDbResolver dbResolver)
+    {
+        this.scopeFactory = scopeFactory;
+        this.dbResolver = dbResolver;
+        
+        RuleFor(req => req.Email)
+            .NotEmpty()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.EmailIsEmpty)
+            .MaximumLength(StringLengths.MediumString)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.EmailIsTooLong)
+            .MustAsync(IsEmailAvailable)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.EmailIsAlreadyTaken)
+            .WithMessage("Email is already taken")
+            .EmailAddress()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.EmailIsInvalid);
+        
+        RuleFor(req => req.Password)
+            .NotNull()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.PasswordIsNull)
+            .SetValidator(new PasswordValidator(RegisterOrganizerRequest.ErrorCodes.PasswordIsTooShort,
+                RegisterOrganizerRequest.ErrorCodes.PasswordIsTooLong,
+                RegisterOrganizerRequest.ErrorCodes.PasswordIsInvalid));
+        
+        RuleFor(req => req.CompanyName)
+            .NotEmpty()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.CompanyNameIsEmpty)
+            .MaximumLength(StringLengths.ShortString)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.CompanyNameIsTooLong);
+        
+        RuleFor(req => req.Address)
+            .NotEmpty()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.AddressIsEmpty)
+            .MaximumLength(StringLengths.MediumString)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.AddressIsTooLong);
+        
+        RuleFor(req => req.TaxId)
+            .NotEmpty()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.TaxIdIsEmpty)
+            .MaximumLength(StringLengths.ShortString)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.TaxIdIsTooLong);
+        
+        RuleFor(req => req.DisplayName)
+            .NotEmpty()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.DisplayNameIsEmpty)
+            .MaximumLength(StringLengths.ShortString)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.DisplayNameIsTooLong);
+        
+        RuleFor(req => req.PhoneNumber)
+            .NotEmpty()
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.PhoneNumberIsEmpty)
+            .MaximumLength(StringLengths.ShortString)
+            .WithCode(RegisterOrganizerRequest.ErrorCodes.PhoneNumberIsTooLong);
+        
+    }
+    
+    private Task<bool> IsEmailAvailable(string email, CancellationToken cancellationToken)
```

Comment:
Add `async` sufix here. I guess it should be also added in `RegisterUserValidator`. Also I'm thinking if we should extract this method somewhere else to make it available for both validators and write it only once. But until this method is small and written only twice both solutions are ok.